netcore/System.Private.CoreLib/src/System.Reflection.Emit/DynamicMethod.cs

```diff
@@ -135,6 +135,8 @@ public DynamicMethod (string name, Type returnType, Type[] parameterTypes, bool
 		[MethodImplAttribute(MethodImplOptions.InternalCall)]
 		private static extern void create_dynamic_method (DynamicMethod m);
 
+		// Clearing of ilgen in create_dynamic_method is not yet synchronized for multiple threads
+		[MethodImpl(MethodImplOptions.Synchronized)]
```

Comment:
Wouldn't be better to move the `mb->ilgen = NULL;` to `CreateDynMethod` ? That way it would be clearer what the flow is and what is the relation between ilgen and mhandle (we could then also possibly lock on ilgen)